solidity_contracts/erc20/contract.sol

```diff
@@ -1520,10 +1530,11 @@ contract BridgeERC20 is Initializable, ERC20Upgradeable, UUPSUpgradeable {
         // ignore mint and burn
         if (from == address(0) || to == address(0)) return;
         if (_isReservedAddress(to)) {
-            require(msg.value >= egressFee, "Bridge: no enough bridge fee");
-
+            require(msg.value == egressFee, "incorrect egress bridge fee");
             // Call bridgeMessage of EVM Runtime
-            bytes memory n_bytes = abi.encodePacked(to, amount, memo);
+            // using abi.encodePacked: only the last field should be of variable length
+            uint32 app_version = 0x653332e5; // sha("bridgeTransferV0(address,uint,string)")
+            bytes memory n_bytes = abi.encodePacked(app_version, to, amount, memo);
```

Comment:
`n_bytes` is a strange name. Maybe `receiver_msg` instead?

Later we will need to signal as part of the `receiver_msg` whether the Solidity contract is assuming it is acting in a one-phase mode (which requires atomicity and is the reason we set the `force_atomic` boolean to true) or it is acting in the two-phase mode.

I suppose we can just assume that `bridgeMsgV0` implies that is acting in one-phase mode. In the future, when we support a two-phase mode as well, we will need to introduce a new version of this message that actually has an explicit argument to indicate whether it is assuming one-phase or two-phase.

Reply:
fixed